src/Project.hs

```diff
@@ -425,3 +439,41 @@ needsTag :: ApprovedFor -> Bool
 needsTag Merge          = False
 needsTag MergeAndDeploy = True
 needsTag MergeAndTag    = True
+
+integrationSha :: PullRequest -> Maybe Sha
+integrationSha PullRequest{integrationStatus = Integrated s _} = Just s
+integrationSha _                                               = Nothing
+
+lookupIntegrationSha :: PullRequestId -> ProjectState -> Maybe Sha
+lookupIntegrationSha pid = integrationSha <=< lookupPullRequest pid
+
+-- | Returns whether the first pull request was approved after the second.
+-- To be used in infix notation:
+--
+-- > pr1 `approvedAfter` pr2
+approvedAfter :: PullRequest -> PullRequest -> Bool
+pr1 `approvedAfter` pr2 = case (mo1, mo2) of
+                          (Just order1, Just order2) -> order1 > order2
+                          _                          -> False
+  where
+  mo1 = approvalOrder <$> approval pr1
+  mo2 = approvalOrder <$> approval pr2
+
+isIntegrated :: IntegrationStatus -> Bool
+isIntegrated (Integrated _ _) = True
+isIntegrated _                = False
+
+isUnfailingIntegrated :: IntegrationStatus -> Bool
+isUnfailingIntegrated (Integrated _ buildStatus) = case buildStatus of
+                                                   BuildPending     -> True
+                                                   (BuildStarted _) -> True
+                                                   BuildSucceeded   -> True
+                                                   (BuildFailed _)  -> False
```

Comment:
I wanted to be explicit and list all cases in case we get a new constructor: the author will have to think about how to actually classify here.  I will leave it as it is for now.

Reply:
`isUnfailedIntegrated` maybe? Or `isNotFailedIntegrated`?